tools/auto_fill_stable_diffusion_cache.py

```diff
@@ -0,0 +1,192 @@
+        compile_command = [
+            "optimum-cli",
+            "export",
+            "neuron",
+            "-m",
+            hf_model_id,
+            "--task",
+            task,
+            "--batch_size",
+            str(batch_size),
+            "--height",
+            str(height),
+            "--width",
+            str(width),
+            "--num_images_per_prompt",
+            str(num_images_per_prompt),
+            "--auto_cast",
+            auto_cast,
+            "--auto_cast_type",
+            auto_cast_type,
+            temp_dir,
+        ]
+        logger.info(f"Running compile command: {' '.join(compile_command)}")
+        try:
+            subprocess.run(compile_command, check=True)
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Failed to compile model: {e}")
+            return
+
+        # Synchronize compiled model to Hugging Face Hub
+        cache_sync_command = ["optimum-cli", "neuron", "cache", "synchronize"]
+        logger.info(f"Running cache synchronize command: {' '.join(cache_sync_command)}")
+
+        try:
+            subprocess.run(cache_sync_command, check=True)
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Failed to synchronize compiled model: {e}")
+            return
+
+    # Log time taken
+    logger.info(f"Compiled and cached model {hf_model_id} w{time.time() - start:.2f} seconds")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Compile and cache a model to the Hugging Face Hub.")
+    parser.add_argument("--hf_model_id", type=str, help="Hugging Face model ID to compile.")
+    parser.add_argument("--task", type=str, help="Task for compilation.")
+    parser.add_argument("--batch_size", type=int, help="Batch size for compilation.")
+    parser.add_argument("--height", type=int, help="Image height for compilation.")
+    parser.add_argument("--width", type=int, help="Image width for compilation.")
+    parser.add_argument(
+        "--num_images_per_prompt", type=int, help="Number of images generated per prompt for compilation."
+    )
+    parser.add_argument(
+        "--auto_cast",
+        type=str,
+        choices=["none", "matmul", "all"],
+        help="Cast operations from FP32 to lower precision to speed up the inference.",
+    )
+    parser.add_argument(
+        "--auto_cast_type", type=str, choices=["bf16", "fp16", "tf32"], help="Auto cast type for compilation."
+    )
+    parser.add_argument("--hf_token", type=str, help="Hugging Face token for authentication if not logged in.")
+    parser.add_argument("--config_file", type=str, help="Path to a json config file with model configurations.")
+    args = parser.parse_args()
+
+    # Ensure either HF token is provided or user is already logged in
+    if args.hf_token:
+        logger.info(f"Logging in to Hugging Face Hub with {args.hf_token[:10]}...")
+        login(args.hf_token)
```

Comment:
This will overwrite the token set on the system if it exists. It would be nice to restore it afterwards.

Reply:
Will add!